addons/crm_addons/controllers/main.py

```diff
@@ -0,0 +1,566 @@
+                    })
+                partner.parent_id = companies[0].id
+                # Enrich the company of the sender
+                self._iap_enrich_partner_from_response(request.env, iap_data, partner.parent_id)
+
+            # Enrich the sender of the email
+            self._iap_enrich_partner_from_response(request.env, iap_data, partner)
+
+            # Gather additional information on the company.
+            if iap_data.get("employees"):
+                enriched_company["employees"] = iap_data.get("employees")
+            annual_revenue = iap_data.get("annual_revenue")
+            if annual_revenue and annual_revenue > 0:
+                enriched_company["annual_revenue"] = annual_revenue
+            elif iap_data.get("estimated_annual_revenue"):
+                enriched_company["estimated_revenue"] = iap_data.get("estimated_annual_revenue")
+            if iap_data.get("company_type"):
+                enriched_company["type"] = iap_data.get("company_type")
+            if iap_data.get("tag"):
+                enriched_company["markets"] = ', '.join(iap_data.get("tag"))
+
+            response['partners'].append({
+                'id': partner.id,
+                'name': partner.name,
+                'address': {
+                    'street': partner.street,
+                    'city': partner.city,
+                    'zip': partner.zip,
+                    'country': partner.country_id.name
+                },
+                'title': partner.function,
+                'phone': partner.phone,
+                'mobile': partner.mobile,
+                'email': partner.email,
+                'image': partner.image_128,
+                'company': {
+                    'id': partner.parent_id.id,
+                    'name': partner.parent_name,
+                    'website': partner.parent_id.website,
+                    'additional_information': enriched_company
+                },
+                'leads': self._get_leads_for_partner_id(request, partner.id)
+            })
+
+        return response
+
+    @http.route('/mail_client_extension/log_single_mail_content', type="json", auth="none", csrf=False, cors="*")
+    def log_single_mail_content(self, **kw):
+        if not CrmAddonsController._authorize(request):
+            return {'authorized': False}
+
+        partner_ids = request.jsonrequest.get("partner_ids")
+        lead_ids = request.jsonrequest.get("lead_ids")
+        message = request.jsonrequest.get("message")
+
+        author_id = int(request.env.user)
+
+        partners = request.env['res.partner'].with_user(request.env.user).browse(int(idStr) for idStr in partner_ids)
+        leads = request.env['crm.lead'].with_user(request.env.user).browse(int(idStr) for idStr in lead_ids)
+
+        errors = []
+        for p in partners:
+            try:
+                p._message_log(body=message, author_id=author_id)
+            except:# Exception as e:
```

Comment:
Is there a reason for `_message_log` to raise random exceptions? Existing callsites don't seem to do any sort of error recovery there (/cc @tde-banana-odoo).

Plus if the issue is a database problem we've blown the cursor so the next log attempt will also break.

And an exception *should* result in a json-rpc error being returned to the caller, which seems more sensible than putting a custom error inside a success?